### DISCUSSION TITLE
Protected struct allowed in static class

### DISCUSSION THREAD

**Discussion by 5cover** [dotnet/csharplang#5266] [Q&A] [answered]
Hello,
```c#
static class foo
{
    protected void doThing() // CS1057
    {

    }
}
```
This code generates the [CS1057 ](https://docs.microsoft.com/dotnet/csharp/misc/cs1057) compiler error, as static classes don’t support inheritance.

So why is the following allowed?
```c#
static class foo
{
    protected struct bar
    {

    }
}
```

**Comment by svick** [comment, marked as answer]
Looking at [the relevant code in the compiler](https://github.com/dotnet/roslyn/blob/67d2d8a/src/Compilers/CSharp/Portable/Symbols/Source/SourceMemberContainerSymbol.cs#L2185), this seems to be a bug that was intentionally preserved from the previous version of the compiler to maintain compatibility.

**Reply by MaceWindu** [reply]
BTW, why it wasn't hidden behind `<Features>strict</Features>`?

**Reply by RikkiGibson** [reply]
It might be reasonable to include this scenario in a warning wave.
